File: fibonacci.rb
def iterative_fib(num)
  index = num
  fib1 = 0
  fib2 = 1
  index.times do 
    fibo = fib1 + fib2
    fib1 = fib2
    fib2 = fibo
  end
    puts fib1
end

def recursive_fib(num)
  if num < 2
    num
  else
    recursive_fib(num - 1) + recursive_fib(num - 2)
  end
end

require 'benchmark'
num = 35
Benchmark.bm do |x|
  x.report("recursive_fib") { recursive_fib(num) }
  x.report("iterative_fib")  { iterative_fib(num)  }
end
    

File: fibonacci.js
var readlineSync = require('readline-sync');

var digit = readlineSync.question("Enter number of digits: ");
var f1 = 0
var f2 = 1
var num = index
var iterative_fib = function(index) {
  index.times do;
  fibo = f1 + f2;
  f1 = f2;
  f2 = fibo; 
  end};
console.log(fibo);
  
iterative_fib();

require 'benchmark'
num = 35
Benchmark.bm do |x|
  x.report("recursive_fib") { recursive_fib(num) }
  x.report("iterative_fib")  { iterative_fib(num)  }
end